napalm/ios/ios.py

```diff
@@ -1955,6 +1955,37 @@ def get_snmp_information(self):
             snmp_dict['chassis_id'] = snmp_chassis
         return snmp_dict
 
+    def get_users(self):
+        """
+        Returns a dictionary with the configured users.
+        The keys of the main dictionary represents the username.
+        The values represent the details of the user,
+        represented by the following keys:
+
+            * level (int)
+            * password (str)
+            * sshkeys (list)
+
+        *Note: sshkeys on ios is the ssh key fingerprint
+
+        The level is an integer between 0 and 15, where 0 is the
+        lowest access and 15 represents full access to the device.
+        """
+        users = {}
+        command = "show run | section username"
+        output = self._send_command(command)
+        for match in re.finditer(r"^username\s+(?P<username>\S+)\s+(?:privilege\s+(?P<priv_level>\S+)\s+)?(?:secret \d+\s+(?P<pwd_hash>\S+))?$", output, re.M):
+            users[match.groupdict()["username"]] = {
+                'level': int(match.groupdict()["priv_level"]) if match.groupdict()["priv_level"] else 1,
+                'password': match.groupdict()["pwd_hash"] if match.groupdict()["pwd_hash"] else "",
+                'sshkeys': []
+            }
+        for match in re.finditer(r"^\s+username\s+(?P<username>\S+)(?P<keys>(?:\n\s+key-hash\s+(?P<hash_type>\S+)\s+(?P<hash>\S+)(?:\s+\S+)?)+)$", output, re.M):
```

Comment:
@ktbyers I couldn't figure out a way to match the key-hash lines multiple times so this is the best I could come up with.